src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,97 @@
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+  handleChange = (key, value) => this.setState({ [key]: value });
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+  render() {
+    const { todos, selectUser } = this.props;
+    const { searchLine, filterBy } = this.state;
+    let filteredTodos = todos.filter((todo) => {
+      if (todo.title === null) {
+        return false;
+      }
+
+      return todo.title.toLowerCase().includes(searchLine.toLowerCase());
+    });
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+    if (filterBy !== 'all') {
+      filteredTodos = filteredTodos.filter(todo => (filterBy === 'active'
+        ? !todo.completed
+        : todo.completed));
+    }
 
-          <button
-            className="TodoList__user-button button"
-            type="button"
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+        <div className="TodoList__list-container">
+          <input
+            type="text"
+            placeholder="Search by title"
+            value={this.state.searchLine}
+            onChange={(event) => {
+              this.setState({ searchLine: event.target.value });
+            }}
+          />
+          <select
+            name="filterBy"
+            value={filterBy}
+            onChange={e => this.handleChange(e.target.name, e.target.value)}
           >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+            <option value="all">all</option>
+            <option value="active">active</option>
+            <option value="completed">completed</option>
+          </select>
+          <ul className="TodoList__list">
+            {filteredTodos.map(todo => (
+              <li
+                className={`
+                  TodoList__item
+                  TodoList__item${todo.completed
+                  ? '--checked'
+                  : '--unchecked'}`
+                  }
+                key={todo.id}
+              >
+                <label>
+                  <input type="checkbox" checked={todo.completed} readOnly />
+                  <p>{todo.title}</p>
+                </label>
+                <button
+                  className="
+                    TodoList__user-button
+                    TodoList__user-button--selected
+                    button"
+                  type="button"
+                  value={todo.userId}
+                  onClick={e => selectUser(+(e.target.value))}
```

Comment:
Назв змінних з однією буквою не може існувати
```suggestion
                  onClick={e => selectUser(Number(e.target.value))}
```